Libraries/LibCrypto/Curves/EdwardsCurve.cpp

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2025, Altomani Gianluca <altomanigianluca@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/ScopeGuard.h>
+#include <LibCrypto/Curves/EdwardsCurve.h>
+#include <LibCrypto/OpenSSL.h>
+
+#include <openssl/core_names.h>
+#include <openssl/evp.h>
+
+namespace Crypto::Curves {
+
+ErrorOr<ByteBuffer> EdwardsCurve::generate_private_key()
```

Comment:
For key material, perhaps it's finally time to revive our RAII secure_zero'd string idea? <https://github.com/SerenityOS/serenity/blob/e10d5a4a232d5e0005315e239753b920bb142d18/Userland/Libraries/LibCore/SecretString.cpp#L46> I believe I mentioned the idea of reviving that class on another PR recently.

Reply:
Should I make an issue about this? I feel like there are a lot of places where this would need to be adjusted.

Reply:
Yeah probably. It will probably be a pretty invasive change to libcrypto API users